packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,276 @@
+| `visual` | `Link, Action, Icon` | `undefined` | Trailing visual element placed on right side of the header |          |
+
+### `BottomSheetFooterLeading` API
+
+| Prop     | Type            | Default     | Description                          | Required |
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the BottomSheet footer      | ✅       |
+| `prefix` | `IconComponent` | `undefined` | Prefix icon placed before title text |          |
+
+### `BottomSheetFooterTrailing` API
+
+| Prop      | Type                                     | Default     | Description                                | Required |
+| --------- | ---------------------------------------- | ----------- | ------------------------------------------ | -------- |
+| `actions` | `{ primary: Action, secondary: Action }` | `undefined` | Renders a primary/secondary action buttons |          |
+
+```ts
+type Action = {
+  onClick: () => void;
+  text: string;
+  type?: 'button' | 'reset' | 'submit';
+  accessibilityLabel: string;
+  isLoading: boolean;
+  isDisabled: boolean;
+  icon: React.ReactNode;
+  iconPosition: 'left' | 'right';
+};
+```
+
+## Composition with Dropdown
+
+We will export `BottomSheet` component separately as an independant component but generally the pattern will be to use it with SelectInput, where in mobile devices the Select's dropdown will be replaced by the BottomSheet.
+
+There are two approaches to doing it:
+
+1. We coupled the BottomSheet & SelectInput tightly and internally conditionally switch the components
+
+Pros:
+
+- Easy implementation for user's end
+- User's don't have to think about breakpoints, conditional rendering etc
+- From blade side, we will have greater control over the pattern
+
+Cons:
+
+- Bundle size will be impacted, even if users are desktop they will get the bundle of BottomSheet (vice versa)
+
+2. We expose BottomSheet indepandantly and let users lazy load the component as needed
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Select as per their needs.
+
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+### Composition Example
+
+```jsx
+import { Spinner, useTheme, useBreakpoint } from "@razorpay/blade";
+
+const BottomSheet = React.lazy();
+const DropdownOverlay = React.lazy();
+
+const App = () => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const isMobile = matchedDeviceType === 'mobile';
+
+  return (
+    <Dropdown selectionType={selectionType}>
+      <SelectInput label="Select Action" />
+      {/* We can either put a fallback spinner or show skeleton loaders */}
+      <React.Suspense fallback={<Spinner />}>
```

Comment:
This will look pretty awkward if the bundle takes time to load 🥲

Reply:
As discussed in the notion doc, from our end we won't do any code splitting if user wants it they can do it, though we won't recommend doing it because bundle size savings either way isn't that huge. Like in storybook examples and docs we will only recommend the way without lazy loading.

Reply:
> because bundle size savings either way isn't that huge

This seems contrasting with the recommendation above favoring approach (2) over (1) because of bundle size.

Reply:
Should be fine anyway we are not recommending this, just a contrived example that it "could be possible" but at the end of the day what we want is to keep dropdown & bottomsheet separate component which we've done.